qmi/instruments/bristol/fos.py

```diff
@@ -1,98 +1,248 @@
 """
 Instrument driver for the four-channel Bristol Fiber-Optic Switch (FOS)
-
-.. autoclass:: Bristol_FOS
-   :members:
-   :undoc-members:
 """
 
+import logging
+import sys
 import typing
 from typing import Optional
 
+import qmi.core.exceptions
 from qmi.core.context import QMI_Context
 from qmi.core.instrument import QMI_Instrument
 from qmi.core.rpc import rpc_method
 
-# Lazy import of the "uldaq" module. See the function _import_modules() below.
+# Lazy import of the "uldaq" or "ul" and "enums" modules. See the function _import_modules() below.
+uldaq = None
+ul, enums = None, None
 if typing.TYPE_CHECKING:
-    import uldaq
-else:
-    uldaq = None
+    if sys.platform.startswith("linux") or sys.platform == "darwin":
+        import uldaq  # type: ignore
+
+    if sys.platform.startswith("win"):
+        import ul  # type: ignore
+        import enums  # type: ignore
+
+
+# Global variable holding the logger for this module.
+_logger = logging.getLogger(__name__)
 
 
 def _import_modules() -> None:
-    """Import the "uldaq" library.
+    """Import the "uldaq" library or "mcculw" modules.
 
-    This import is done in a function, instead of at the top-level,
-    to avoid an unnecessary dependency for programs that do not access
-    the instrument directly.
+    This import is done in a function, instead of at the top-level, to avoid an unnecessary
+    dependencies for programs that do not access the instrument directly.
     """
-    global uldaq
-    if uldaq is None:
-        import uldaq  # pylint: disable=W0621
+    global uldaq, ul, enums
+    _logger.debug("Importing %s modules", sys.platform)
+    if (sys.platform.startswith("linux") or sys.platform == "darwin") and uldaq is None:
+        import uldaq  # type: ignore
 
+    elif sys.platform.startswith("win") and (ul is None or enums is None):
+        from mcculw import ul, enums  # type: ignore
```

Comment:
when importing earlier for typing checking `ul` and `enums` are imported but here they are imported from `mcculw`. Should we do the same for type checking?